File: LinkedList/SinglyLinkedList/SinglyLinkedList.java
import java.util.Scanner;

class ListNode{
    public int data;
    public ListNode next;

    ListNode(int data){
        this.data = data;
        this.next = null;
    }
}

public class SinglyLinkedList{

    public static void print(ListNode head){
        ListNode curr = head;
        while(curr != null){
            System.out.print(curr.data + " --> ");
            curr = curr.next;
        }
        System.out.print(" null ");
    }
    public static void main(String[] args) {
        
        ListNode head = null;
        ListNode curr = null;
        Scanner sc = new Scanner(System.in);
        while(true){

            System.out.println("Want to enter elemnent in linked list?");
            System.out.print("Enter 1 for yes or -1 for no : ");
            int num = sc.nextInt();

            if(num == 1){
                System.out.print("Enter node value : ");
                int data = sc.nextInt();
                ListNode node = new ListNode(data);
                if(head == null){
                    head = node;
                    curr = node;
                }
                else{
                    curr.next = node;
                    curr = node;
                }
            }
            else{
                break;
            }
        }

        System.out.println("Singly Linked List");
        print(head);

        sc.close();
    }
}

File: LinkedList/SinglyLinkedList/SinglyLinkedList.cpp
#include<bits/stdc++.h>
using namespace std;

struct ListNode{
    int data;
    ListNode *next;

    ListNode(int data){
        this->data = data;
        this->next = nullptr;
    }
};

void print(ListNode *head){
    ListNode *curr = head;
    while(curr != nullptr){
        cout << curr->data << " --> ";
        curr = curr->next;
    }
    cout << "nullptr" << endl;
}

int main(){

    ListNode *head = nullptr;
    ListNode *curr = nullptr;
    
    while(true){

        cout << "Want to enter elemnent in linked list?" << endl;
        cout << "Enter 1 for yes or -1 for no : ";

        int num;
        cin >> num;
        if(num == 1){
            int data;
            cout << "Enter node value : ";
            cin >> data;

            ListNode *node = new ListNode(data);
            if(head == nullptr){
                head = node;
                curr = node;
            }
            else{
                curr->next = node;
                curr = curr->next;
            }
        }
        else{
            break;
        }
    }

    cout << "Singly Linked List" << endl;
    print(head);
}